src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -302,5 +361,95 @@ std::string ObjectCacheStore::get_cache_file_path(std::string cache_file_name,
   return m_cache_root_dir + cache_file_dir + cache_file_name;
 }
 
+void ObjectCacheStore::handle_throttle_ready(uint64_t tokens, uint64_t type) {
+  m_io_throttled.store(false);
+  std::lock_guard lock(m_throttle_lock);
+  if (type & ROC_QOS_IOPS_THROTTLE){
+    m_iops_tokens += tokens;
+  } else if (type & ROC_QOS_BPS_THROTTLE){
+    m_bps_tokens += tokens;
+  } else {
+    lderr(m_cct) << "unknow throttle type." << dendl;
+  }
+}
+
+bool ObjectCacheStore::take_token_from_throttle(uint64_t object_size,
+                                                uint64_t object_num) {
+  if (m_io_throttled.load() == true) {
+    return false;
+  }
+
+  int flag = 0;
+  bool wait = false;
+  if (!wait && (m_qos_enabled_flag & ROC_QOS_IOPS_THROTTLE)) {
+    std::lock_guard lock(m_throttle_lock);
+    if (object_num < m_iops_tokens) {
```

Comment:
`m_iops_tokens` is initialized to zero so this will always fail the test on the first invokation, which then hits line 391 which subtracts 1 from 0 (to get a very large positive number).